createst.py

```diff
@@ -141,10 +141,29 @@ def write_to_file(data):
         sys.exit(1)
     with open(test_yaml_path, "w+") as fp:
         fp.write("# *** Add configuration here ***\n\n")
-        if not args["strictcsums"]:
-            fp.write("args:\n- -k none\n\n")
+        if check_set_args():
+            fp.write("args:\n")
+            if not args["strictcsums"]:
+                fp.write("- -k none\n")
+            if args["midstream"]:
+                fp.write("- --set stream.midstream=true\n")
+            fp.write("\n")
         fp.write(data)
 
+def check_set_args():
+    """
+    Check if the user wants midstream set to true and/or to have strict
+    checksums
+    """
+    features = ["midstream", "strictcsums"]
+    no_features = True
+    for item in features:
+        if args[item]:
+            if item != "strictcsums":
+                return True
+            no_features = False 
+        elif no_features:
+            return True
```

Comment:
hmm seems like this acts weird for `strictcsums`. I see an empty `args:` section.
I see that one option working opposite of others may be complicating things. Sorry about that.
So, we'll have to make a special handling for `strictcsums` but it can be done a bit differently. Took me a while but I have (on top of your v2),

```python
diff --git a/createst.py b/createst.py
index 05f4318..4ddbbe6 100755
--- a/createst.py
+++ b/createst.py
@@ -143,18 +143,21 @@ def write_to_file(data):
         fp.write("# *** Add configuration here ***\n\n")
         if check_set_args():
             fp.write("args:\n")
-            if not args["strictcsums"]:
-                fp.write("- -k none\n")
-            if args["midstream"]:
-                fp.write("- --set stream.midstream=true\n")
-            fp.write("\n")
+        if not args["strictcsums"]:
+            fp.write("- -k none\n")
+        if args["midstream"]:
+            fp.write("- --set stream.midstream=true\n")
+        fp.write("\n")
         fp.write(data)
 
 def check_set_args():
-    features = ["midstream"]
+    features = ["midstream", "strictcsums"]
     for item in features:
         if args[item]:
+            if item == "strictcsums":
+                return False
             return True
+    return True
 
 def test_yaml_format(func):
     """
```

Please note that I have only taken code out of the scope of ` if check_set_args():` as it's only `args:` that really get affected by this. There are no other changes in that function.

Leaving a case table for you to check against (or point out mistakes, if there are any):

Case I: No args
Result: Should have `-k none`

Case II: `--strictcsums` only
Result: Should have no `args` section

Case III: `--midstream` only
Result: Should have `-k none` as well as `- --set stream.midstream=true`

Case IV: `--midstream` + `--strictcsums`
Result: Should only have `- --set stream.midstream=true`

Reply:
Thanks a lot for taking your time to walk me through this >__<' 
I swear that I tested all cases yesterday, but I understand what happened with this solution of mine. The order of the elements in the array impacts in its results.
I wanted to check if I was able to come up with another working solution as well, so I'm sharing that one in a new PR now.